src/main/java/jenkins/plugins/logstash/persistence/ElasticSearchDao.java

```diff
@@ -106,7 +110,36 @@ public ElasticSearchDao(URI uri, String username, String password) {
       auth = null;
     }
 
-    clientBuilder = factory == null ? HttpClientBuilder.create() : factory;
+    clientBuilder = factory;
+  }
+
+  private byte[] getKeystoreBytes() {
+    return keystoreBytes;
+  }
+
+  private String getKeyStorePassword() {
+    return keyStorePassword;
+  }
+
+  private synchronized HttpClientBuilder getClientBuilder() throws IOException {
```

Comment:
I'm not sure why do we need to change the client/cert handling for this?

Reply:
We need to change because this class must be serializable. As clientBuilder is not serializable it has to be transient and must be recreated upon first use

Reply:
right, thanks for the explanation